pkg/domain/infra/abi/images.go

```diff
@@ -54,13 +63,15 @@ func (ir *ImageEngine) Prune(ctx context.Context, opts entities.ImagePruneOption
 		WithSize:                true,
 	}
 
-	if !opts.All {
+	if !opts.All && !filterPrefix("dangling=", pruneOptions.Filters) {
 		pruneOptions.Filters = append(pruneOptions.Filters, "dangling=true")
 	}
-	if opts.External {
-		pruneOptions.Filters = append(pruneOptions.Filters, "containers=external")
-	} else {
-		pruneOptions.Filters = append(pruneOptions.Filters, "containers=false")
+	if !filterPrefix("containers=", pruneOptions.Filters) {
+		if opts.External {
+			pruneOptions.Filters = append(pruneOptions.Filters, "containers=external")
+		} else {
+			pruneOptions.Filters = append(pruneOptions.Filters, "containers=false")
+		}
```

Comment:
I think this should be removed, @vrothberg correct?

Reply:
@Luap99 I think these are dangling images from `multistage` builds. I dont think API should return these. But I can check what docker is returning.

Reply:
I mean the change should be removed, see https://github.com/containers/common/pull/879/commits/134e83ff55ce83a37756ef473bf083b65e2c2c3c
@vrothberg fixed the problem in c/common so this is no longer needed

Reply:
Ah i see. Fair if that is fixed in upstream then check could be removed.

Reply:
Yes, that's not needed anymore.